### ISSUE TITLE
Crash after adding text area and clicking on the any other tool.

### ISSUE THREAD

**Issue by vanheck** [flameshot-org/flameshot#2283]
### Flameshot Version

```sh
$ dnf list installed | grep flameshot
flameshot.x86_64                                   11.0.0-1.fc35                          @updates  

$ flameshot --version
Flameshot v11.0.0 (-)
Compiled with Qt 5.15.2
```

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Fedora 35 + Gnome 41.3 (wayland)

### Description

Crash after adding text area and clicking on the any other tool.

### Steps to reproduce

1. `flameshot gui`
2. select area to take the screen
3. Click on the button Add text to your capture (T) - write some text
4. Click on any other tool to anotate (line, arrow line, rectangle, etc.) - crash

### Screenshots or screen recordings

https://youtu.be/X5rv4BtETr8

### System Information

```sh
$ cat /etc/os-release
NAME="Fedora Linux"
VERSION="35 (Workstation Edition)"
ID=fedora
VERSION_ID=35
VERSION_CODENAME=""
PLATFORM_ID="platform:f35"
PRETTY_NAME="Fedora Linux 35 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:35"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f35/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=35
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=35
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

![obrazek](https://user-images.githubusercontent.com/9953249/150100983-a15880df-baba-4a8b-afbc-c68e8b78ce5b.png)

```sh
$ uname -a
Linux jvh-machine 5.15.14-200.fc35.x86_64 #1 SMP Tue Jan 11 16:49:27 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by vanheck** [comment]
Not sure if its the right command, but here is some locale settings:
```sh
$ locale
LANG=cs_CZ.UTF-8
LC_CTYPE="cs_CZ.UTF-8"
LC_NUMERIC="cs_CZ.UTF-8"
LC_TIME="cs_CZ.UTF-8"
LC_COLLATE="cs_CZ.UTF-8"
LC_MONETARY="cs_CZ.UTF-8"
LC_MESSAGES="cs_CZ.UTF-8"
LC_PAPER="cs_CZ.UTF-8"
LC_NAME="cs_CZ.UTF-8"
LC_ADDRESS="cs_CZ.UTF-8"
LC_TELEPHONE="cs_CZ.UTF-8"
LC_MEASUREMENT="cs_CZ.UTF-8"
LC_IDENTIFICATION="cs_CZ.UTF-8"
LC_ALL=
```

**Comment by mmahmoudian** [comment]
I can reproduce this in a clean VM

**Comment by mllanes** [comment]
I can verify it happens on Pop !OS 21.10 as described using kernel 5.15.11-76051511-generic: Selecting any tool after having selecting the text input closes the selection tool.

**Comment by AndreaMarangoni** [comment]
> @mmahmoudian did you set any locales? On arch I dont have this issue.

Interesting. On my arch, with Uk locale, I had to change locale settings on [filenamehandler.cpp](https://github.com/flameshot-org/flameshot/pull/2238/files#r789925812) to make it work. (I know I should have reported it)

**Comment by AndreaMarangoni** [comment]
> I can verify it happens on Pop !OS 21.10 as described using kernel 5.15.11-76051511-generic: Selecting any tool after having selected the text input closes the selection tool.
> 
> No errors throw, here the CLI output:
> 
> ```
> ➜  flameshot gui
> flameshot: info: Screenshot aborted.
> ```
> 
> **Note**, if you "de-select" the text tool before selecting another tool it **does not crashes** on my system.

and basically I came here to report this.. if you select `Text tool` and then any other tool it crashes.

**Comment by borgmanJeremy** [comment]
@AndreaMarangoni can you please make that change a separate PR. I can get that merged into 11.1 on its own if it solves the issue.

**Comment by AndreaMarangoni** [comment]
> @AndreaMarangoni can you please make that change a separate PR. I can get that merged into 11.1 on its own if it solves the issue.

Not sure I understood correctly, but the fix for the locale and the crash after selecting the `Text tool` are two separate issues.
I can push a PR with the locale fix anytime.

**Comment by AndreaMarangoni** [comment]
@borgmanJeremy FYI #2304

**Comment by Drarig29** [comment]
I confirm that it doesn't crash if you first un-select the text area tool, and select the other tool.

**Comment by yutv** [comment]
It woks when you initiate UI from the flameshot main menu
![image](https://user-images.githubusercontent.com/1616522/152584037-fb3eb69e-210e-45b7-8843-d9cdb392842e.png)

but doesn't work in case of using a hotkey, see https://github.com/flameshot-org/flameshot/issues/2353#issuecomment-1030232891

**Comment by AndreaMarangoni** [comment]
I found and fixed the issue. PR soon.

**Comment by AndreaMarangoni** [comment]
PR #2369

**Comment by borgmanJeremy** [comment]
Resolved with #2369
